### PR TITLE
Add Testrail Unit Tests, Cron, and Slack Notifs

### PR DESCRIPTION
**Pull request by jjSDET**
Adding unit tests for the testrail apis used in `taskcluster` tasks in android `release-notify-testrail`.

Adding Github Action for daily cron to track status by posting results in `mobile-test-sandbox` slack channel.

### CODE REVIEW

File: .github/workflows/testrail-api-tests.yml [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      id: slack-success
+      uses: slackapi/slack-github-action@v1.26.0
+      with:
+        channel-id: ${{ secrets.SLACK_MOBILE_ALERTS_SANDBOX_CHANNEL }}
+        payload: |
+          {
+            "type": "mrkdwn",
+            "text": "TestRail API tests passed! :tada:\n\n*Workflow:* ${{ github.workflow }}\n*Job:* ${{ github.job }}\n\n*Status:* ${{ job.status }}\n*Test results:* Passed\n\nLink to results: <${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}|View results>"
+          }
+
```
Comment (AaronMT) [inline .github/workflows/testrail-api-tests.yml:65]: newline

File: .github/workflows/testrail-api-tests.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          }
+        
+
+    - name: Send Slack message on test success
+      if: ${{ success() }}
+      id: slack-success
+      uses: slackapi/slack-github-action@v1.26.0
+      with:
+        channel-id: ${{ secrets.SLACK_MOBILE_ALERTS_SANDBOX_CHANNEL }}
+        payload: |
```
Comment (AaronMT) [inline .github/workflows/testrail-api-tests.yml:60]: Same here. Can probably specify a success template vs a failure template payload file too if needed.

File: testrail/requirements.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+requests
```
Comment (AaronMT) [inline testrail/requirements.txt:1]: newline

File: testrail/tests/testrail_api_tests.py [hunk 4]
```diff
@@ -0,0 +1,413 @@
+from datetime import datetime
```
Comment (AaronMT) [inline testrail/tests/testrail_api_tests.py:1]: MPL

**Comment by AaronMT** [comment]
Remember to squash commits too

File: .github/workflows/testrail-api-tests.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+      if: ${{ failure() }}
+      id: slack-failure
+      uses: slackapi/slack-github-action@v1.26.0
+      with:
+        channel-id: ${{ secrets.SLACK_MOBILE_ALERTS_SANDBOX_CHANNEL }}
+        payload: |
+          {
+            "type": "mrkdwn",
+            "text": "TestRail API tests failed! :tada:\n\n*Workflow:* ${{ github.workflow }}\n*Job:* ${{ github.job }}\n\n*Status:* ${{ job.status }}\n*Test results:* Failed\n\nLink to results: <${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}|View results>"
+          }
```
[the remark this answers is not in the document]
Comment (jjSDET) [reply .github/workflows/testrail-api-tests.yml:51]: i tried but couldn't get that to work using `slackapi`. i'll give it another try to beautify the message templating.

[the remark this answers is not in the document]
Comment (jjSDET) [reply .github/workflows/testrail-api-tests.yml:51]: ah, i see. i was using an older version that didn't properly support slack blocks...

File: .github/workflows/testrail-api-tests.yml (hunk 2 above)
Comment (jjSDET) [reply .github/workflows/testrail-api-tests.yml:59]: yeah, forgot to add one. figure we can add a template when we start rolling them out in mobile-ci-messaging stuffs?

**Pull request merged**